src/mem_table.rs

```diff
@@ -170,16 +178,24 @@ impl KVTable {
 
     /// Puts a value, returning as soon as the value is written to the memtable but before
     /// it is flushed to durable storage.
-    fn put(&self, key: &[u8], value: &[u8]) {
+    fn put(&self, key: &[u8], value: &[u8], attrs: RowAttributes) {
         self.map.insert(
             Bytes::copy_from_slice(key),
-            ValueDeletable::Value(Bytes::copy_from_slice(value)),
+            MemValue {
+                value: ValueDeletable::Value(Bytes::copy_from_slice(value)),
+                attrs,
+            },
         );
     }
 
-    fn delete(&self, key: &[u8]) {
-        self.map
-            .insert(Bytes::copy_from_slice(key), ValueDeletable::Tombstone);
+    fn delete(&self, key: &[u8], attrs: RowAttributes) {
+        self.map.insert(
+            Bytes::copy_from_slice(key),
+            MemValue {
+                value: ValueDeletable::Tombstone,
+                attrs,
+            },
+        );
```

Comment:
in my understanding, the tombstone might worths to be kept in the key part before inserting it into memtable.

if it's placed in the value part, every key will only have one latest value. this will have issue when we have multiple snapshots running at the same time. one snapshot may reference a key "key1" when it started at sequence number "100", and later "key1" is deleted at sequence "104", these two "key1" at different time have to be both existed in the memtable.

Reply:
this is a good call, I'll make that change

Reply:
hmm, I think we should leave this for the future when we implement snapshot isolation in the memtable since this would require introducing yet another type for just (key + isTombstone) instead of just using (key + attributes) since you don't know the attributes ahead of time when you search for a key. we can do this in a follow up as this PR is already rather large.

Reply:
looks cool, +1 for keeping this PR self-contained, let's plan to make keys multi-version support in another self-contained change. i can try working on it over the weekend. :)

Reply:
added an issue on this: https://github.com/slatedb/slatedb/issues/274